puffin/src/scope_details.rs

```diff
@@ -0,0 +1,212 @@
+use std::{borrow::Cow, collections::HashSet, sync::Arc};
+
+use parking_lot::RwLock;
+
+use crate::{clean_function_name, fetch_add_scope_id, short_file_name, ScopeId};
+
+#[derive(Default, Clone)]
+struct InnerDetails {
+    pub(crate) scope_id_to_details: std::collections::HashMap<ScopeId, ScopeDetailsOwned>,
+    pub(crate) string_to_scope_id: std::collections::HashMap<String, ScopeId>,
+}
+
+/// Provides read access to scope details.
+#[derive(Default, Clone)]
+pub struct ScopeDetails(
```

Comment:
This is confusing naming.

`ScopeDetailsOwned`, `ScopeDetailsRef`, `CustomScopeDetails` is all about the details of a _single_ profile scope.

…but `ScopeDetails` and `InnerDetals` is the collection of all scopes.

Reply:
This is also an argument against `ScopeDetails` as a name: it can't be pluralized. "The details for one scope vs the details of many scopes". Better then with `ScopeInfo`: "The info of one scope vs the info**s** for many scopes.

So I would suggest`ScopeInfo` for info about a single scope, and then `AllScopeInfos` (to be super explicit) for this collection of all infos we have here.